src/backend/src/services/projects.services.ts

```diff
@@ -123,7 +140,177 @@ export default class ProjectsService {
           }
         }
       },
-      include: { project: true, changes: true }
+      include: {
+        project: { include: { goals: true, features: true, otherConstraints: true } },
+        changes: true,
+        links: true
+      }
+    });
+    const { wbsElementId } = createdWbsElement;
+
+    // const createdProject = createdWbsElement.project;
+    const createdProject = await prisma.project.findUnique({
+      where: { wbsElementId },
+      include: { goals: true, features: true, otherConstraints: true }
+    });
+    // if it doesn't exist we error
+    if (!createdProject) throw new NotFoundException('Project', wbsElementId);
+
+    /* FROM UPDATE PROJECT METHOD */
+    const budgetChangeJson = createChange(
+      'budget',
+      createdProject.budget,
+      budget,
+      crId,
+      user.userId,
+      createdWbsElement.wbsElementId
+    );
+
+    const summaryChangeJson = createChange('summary', createdProject.summary, summary, crId, userId, wbsElementId);
+
+    const projectManagerChangeJson = createChange(
+      'project manager',
+      await getUserFullName(createdWbsElement.projectManagerId),
+      await getUserFullName(projectManagerId),
+      crId,
+      userId,
+      wbsElementId
+    );
+
+    const projectLeadChangeJson = createChange(
+      'project lead',
+      await getUserFullName(createdWbsElement.projectLeadId),
+      await getUserFullName(projectLeadId),
+      crId,
+      userId,
+      wbsElementId
+    );
+
+    let changes = [];
+
+    // add to changes if not undefined
+    if (budgetChangeJson !== undefined) {
+      changes.push(budgetChangeJson);
+    }
+    if (summaryChangeJson !== undefined) {
+      changes.push(summaryChangeJson);
+    }
+    if (projectManagerChangeJson !== undefined) {
+      changes.push(projectManagerChangeJson);
+    }
+    if (projectLeadChangeJson !== undefined) {
+      changes.push(projectLeadChangeJson);
+    }
+
+    // Dealing with lists
+    const rulesChangeJson = createListChanges(
+      'rules',
+      createdProject.rules.map((rule) => {
+        return {
+          element: rule,
+          comparator: rule,
+          displayValue: rule
+        };
+      }),
+      rules
+        ? rules.map((rule) => {
+            return {
+              element: rule,
+              comparator: rule,
+              displayValue: rule
+            };
+          })
+        : [],
+      crId,
+      userId,
+      wbsElementId
+    );
+
+    const goalsChangeJson = createListChanges(
+      'goals',
+      descriptionBulletsToChangeListValues(createdProject.goals),
+      goals ? goals.map((goal) => descriptionBulletToChangeListValue(goal)) : [],
+      crId,
+      userId,
+      wbsElementId
+    );
+
+    const featuresChangeJson = createListChanges(
```

Comment:
ok yes, but you see how u could just copy and paste alot fo this stuff from the edit, just make all this description bullet and link changes and creation shit its own function in a util file and call it in both places